renderdoc/core/resource_manager.h

```diff
@@ -675,6 +709,15 @@ void ResourceManager<Configuration>::MarkResourceFrameReferenced(ResourceId id,
   if(id == ResourceId())
     return;
 
+  if(IsDirtyFrameRef(refType))
+  {
+    Prepare_ResourceIfActivePostponed(id);
+    UpdateLastWriteTime(id);
+
+    if(IsBackgroundCapturing(m_State))
```

Comment:
This return `if` should be outside `if(IsDirtyFrameRef())`, since we don't want to do any reference tracking while in the background either read or write.

At the moment you've only set up mostly write-only references during background capturing but we don't want to have an unexpected problem lurking here if read-only references come through and start leaking.